examples/temporal_graph_learning/gnn_encoder.py

```diff
@@ -0,0 +1,145 @@
+from graphstorm import model as gsmodel
+
+from model_utils import get_unique_nfields
+from model_utils import to_per_field_nfeats
+from model_utils import rel_field_map
+from model_utils import rel_name_map
+from model_utils import average_over_fields
+from model_utils import get_unique_etype_triplet
+from model_utils import get_temporal_restrict_fields
+
+from model_utils import merge_multi_blocks, get_merge_canonical_etype_mapping
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from dgl.nn import GATConv
+
+class RelationalGraphEncoder(gsmodel.gnn_encoder_base.GraphConvEncoder):
+    def __init__(self, g, h_dim, out_dim, num_hidden_layers, num_heads=0, dropout=0):
+        super(RelationalGraphEncoder, self).__init__(h_dim, out_dim, num_hidden_layers)
+        self.node_fields = get_unique_nfields(g.etypes)
+        for _ in range(num_hidden_layers):
+            self.layers.append(
+                RelGraphConvLayer(
+                    g,
+                    in_feat=h_dim,
+                    out_feat=h_dim,
+                    num_heads=num_heads,
+                    dropout=dropout,
+                    activation=nn.LeakyReLU(),
+                )
+            )
+        self.layers.append(
+            RelGraphConvLayer(
+                g,
+                in_feat=h_dim,
+                out_feat=out_dim,
+                num_heads=num_heads,
+                dropout=dropout,
+                activation=nn.LeakyReLU(),
+            )
+        )
+
+    def forward(self, blocks, inputs):
+        h = to_per_field_nfeats(inputs, self.node_fields)
+        for layer, block in zip(self.layers, blocks):
+            h = layer(block, h)
+        h = average_over_fields(h)
+        return h
+
+class RelGraphConvLayer(nn.Module):
```

Comment:
is this RGAT?

Reply:
please provide docstrings for every class.

Reply:
I added docstring to each function

I also update function/class name accordingly.